test/StringToolsTests.h

```diff
@@ -0,0 +1,333 @@
+                   String("11"),DefaultSplitResult[9]);
+        TEST_EQUAL("Split(const_String&,const_String&,const_Whole)-DefaultElement11",
+                   String("parts."),DefaultSplitResult[10]);
+        TEST_EQUAL("Split(const_String&,const_String&,const_Whole)-ISplitCount",
+                   size_t(6),ISplitResult.size());
+        TEST_EQUAL("Split(const_String&,const_String&,const_Whole)-ISplitElement1",
+                   String("Th"),ISplitResult[0]);
+        TEST_EQUAL("Split(const_String&,const_String&,const_Whole)-ISplitElement2",
+                   String("s "),ISplitResult[1]);
+        TEST_EQUAL("Split(const_String&,const_String&,const_Whole)-ISplitElement3",
+                   String("s a str"),ISplitResult[2]);
+        TEST_EQUAL("Split(const_String&,const_String&,const_Whole)-ISplitElement4",
+                   String("ng that w"),ISplitResult[3]);
+        TEST_EQUAL("Split(const_String&,const_String&,const_Whole)-ISplitElement5",
+                   String("ll be spl"),ISplitResult[4]);
+        TEST_EQUAL("Split(const_String&,const_String&,const_Whole)-ISplitElement6",
+                   String("t into 11 parts."),ISplitResult[5]);
+    }//Tokenizing
+
+    {//Pattern Matching
+        String StartsWithSourceString("Autocannibalism");
+        TEST_EQUAL("StartsWith(const_StringView,const_StringView,const_Boole)-Pass",
+                   true,StringTools::StartsWith(StartsWithSourceString,"auto",false));
+        TEST_EQUAL("StartsWith(const_StringView,const_StringView,const_Boole)-CasePass",
+                   true,StringTools::StartsWith(StartsWithSourceString,"Auto",true));
+        TEST_EQUAL("StartsWith(const_StringView,const_StringView,const_Boole)-Fail",
+                   false,StringTools::StartsWith(StartsWithSourceString,"Self",false));
+        TEST_EQUAL("StartsWith(const_StringView,const_StringView,const_Boole)-CaseFail",
+                   false,StringTools::StartsWith(StartsWithSourceString,"AUTO",true));
+
+        String EndsWithSourceString("ThatsTheJoke.jpg");
+        TEST_EQUAL("EndsWith(const_StringView,const_StringView,const_Boole)-Pass",
+                   true,StringTools::EndsWith(EndsWithSourceString,".JPG",false));
+        TEST_EQUAL("EndsWith(const_StringView,const_StringView,const_Boole)-CasePass",
+                   true,StringTools::EndsWith(EndsWithSourceString,".jpg",true));
+        TEST_EQUAL("EndsWith(const_StringView,const_StringView,const_Boole)-Fail",
+                   false,StringTools::EndsWith(EndsWithSourceString,".png",false));
+        TEST_EQUAL("EndsWith(const_StringView,const_StringView,const_Boole)-CaseFail",
+                   false,StringTools::EndsWith(EndsWithSourceString,".JPG",true));
+
+        String FirstSource("Test/Data/Results.txt");
+        String SecondSource("WS_29_06_18.dat");
+        String ThirdSource("Test\\Data\\Results.txt");
+        String FourthSource("*Star*");
+
+        String FirstPattern("Test/Data/*.txt");
```

Comment:
Pattern names that indicate what they match would improve readabilty. This could "PatternMatchesFirst" or "PatternMatchesUnixTextFile" for example.

Reply:
Renamed.